ampd/src/handlers/sui_verify_msg.rs

```diff
@@ -12,3 +29,322 @@ pub struct Message {
     pub source_address: SuiAddress,
     pub payload_hash: Hash,
 }
+
+#[derive(Deserialize, Debug)]
+#[try_from("wasm-messages_poll_started")]
+struct PollStartedEvent {
+    #[serde(rename = "_contract_address")]
+    contract_address: TMAddress,
+    poll_id: PollID,
+    source_gateway_address: SuiAddress,
+    messages: Vec<Message>,
+    participants: Vec<TMAddress>,
+}
+
+pub struct Handler<C, B>
+where
+    C: SuiClient + Send + Sync,
+    B: BroadcasterClient,
+{
+    worker: TMAddress,
+    voting_verifier: TMAddress,
+    rpc_client: C,
+    broadcast_client: B,
+}
+
+impl<C, B> Handler<C, B>
+where
+    C: SuiClient + Send + Sync,
+    B: BroadcasterClient,
+{
+    #[allow(dead_code)]
+    pub fn new(
+        worker: TMAddress,
+        voting_verifier: TMAddress,
+        rpc_client: C,
+        broadcast_client: B,
+    ) -> Self {
+        Self {
+            worker,
+            voting_verifier,
+            rpc_client,
+            broadcast_client,
+        }
+    }
```

Comment:
not necessary, it doesn't do any translation or checks, so you might as well use the object initializer